rmw_connextdds_common/src/common/rmw_publication.cpp

```diff
@@ -166,6 +166,14 @@ rmw_api_connextdds_create_publisher(
     }
   }
 
+  // Adapt any 'best available' QoS options
+  rmw_qos_profile_t adapted_qos_policies = *qos_policies;
+  rmw_ret_t ret = rmw_dds_common::qos_profile_get_best_available_for_topic_publisher(
+    node, topic_name, &adapted_qos_policies, rmw_get_subscriptions_info_by_topic);
```

Comment:
```suggestion
    node, topic_name, &adapted_qos_policies, rmw_api_connextdds_get_subscriptions_info_by_topic);
```

https://github.com/ros2/rmw_connextdds/blob/1252ed842c92385787fb59b15a385e34408b1602/rmw_connextdds_common/include/rmw_connextdds/rmw_api_impl.hpp#L201